dot/network/notifications.go

```diff
@@ -53,8 +53,16 @@ type (
 
 	// NotificationsMessageHandler is called when a (non-handshake) message is received over a notifications stream.
 	NotificationsMessageHandler = func(peer peer.ID, msg NotificationsMessage) (propagate bool, err error)
+
+	// NotificationsMessageBatchHandler is called when a (non-handshake) message is received over a notifications stream in batch processing mode.
+	NotificationsMessageBatchHandler = func(peer peer.ID, msg NotificationsMessage) (batchMsgs []*transactionBatchMessage, err error)
 )
 
+type transactionBatchMessage struct {
+	msg  NotificationsMessage
+	peer peer.ID
+}
```

Comment:
is this for all notifications messages, or only transactions messages?

Reply:
seems like it could be generic, so maybe rename to `batchMessage`?

Reply:
Done